packages/astro/src/core/errors/errors-data.ts

```diff
@@ -80,7 +93,6 @@ export const StaticClientAddressNotAvailable = {
 	title: '`Astro.clientAddress` is not available in static mode.',
 	message:
 		"`Astro.clientAddress` is only available when using `output: 'server'` or `output: 'hybrid'`. Update your Astro config if you need SSR features.",
-	hint: 'See https://docs.astro.build/en/guides/server-side-rendering/#enabling-ssr-in-your-project for more information on how to enable SSR.',
```

Comment:
@Princesseuh Now that each `AstroError` logs a link to the error reference on docs, I updated all hints to remove any that were just calling out 1+ external documentation URLs already linked to from our docs reference page.

Added a note to the README to clarify for the future that `hint` should only be used for suggesting (err... "hinting") a specific solution or fix to the user.

✅ "Does the file exist?"
❌ "See https://docs.astro.build/en/guides/some-docs-that-may-or-may-not-help-you-solve-this for more information"

I think this makes sense given the new error log format (see screenshots in my last comment) but wanted to make sure you saw this and get your thoughts, if any!

Reply:
From @Princesseuh (threading this here to keep the convo going in one place):

> In general, the entire purpose of the hints wasn't necessarily to be short, right. They're supposed to get users to potential solutions as soon as possible, so they should be concise (much like error messages), but not omit information that they can in fact provide.
> 
> Removing links from them to link to the error reference in addition of potentially delaying finding a solution, requires the user to do multiple context switches when we can link to docs page that explain the domain directly.
> 
> I don't personally see the value of reducing the length of errors that much. If I get an error, I'd rather the terminal be spammed with all relevant information than be pretty looking (and I'd much rather resize my terminal than visit multiple webpages)
> 
> We could argue that this new format makes it easier to ingest information, but I don't think that's true because of things being all together vs in different places. It just looks, subjectively, prettier.
> 
> (Edit, oops this was meant to be a reply to the comment, but for some reason GitHub Mobile put it top level)

Reply:
> In general, the entire purpose of the hints wasn't necessarily to be short, right. They're supposed to get users to potential solutions as soon as possible, so they should be concise (much like error messages), but not omit information that they can in fact provide.

That's a great callout. Yes, this PR now explicitly changes the purpose of `hint` to point you to the solution **but not educate you.** That job is left to the error reference URL which is always included in the log, now more prominently than before.

As a reminder, this is what a new error log looks like, with a hint + error reference URL included and clearly visible.

![image](https://github.com/withastro/astro/assets/622227/dbb657ba-fec5-4715-b517-dbe57ed82db8)

The reasoning is that having docs and links in the hint is actually less effective for all users.
- For users who don't need education (ex: they already know what the `alt` attribute is) it's more verbose and adds mental overhead to parse the message for the relevant info you need to solve.
- For the user who does need education, 2+ links is less effective than 1, very clear "go here to learn more" URL. Putting on my new user hat: "wait, this error reference is a URL to learn more but then this hint also includes 1+ different URLs that might help me, which one should i be clicking?"

We should trust that our error reference pages in docs **_are_** the best place to go to learn what the error is and how to solve it. If they aren't, then right solution should be to improve them vs. working around them by adding different additional URLs to the error log.

Reply:
I see your point, but I (respectfully) don't agree. 

Of course, it's on a error to error basis right, I can probably agree to remove some of the ones that are just links, but I don't agree with generally always shortening the hints and removing the information.

The `alt` one is the most telling to me: I do not mean to exaggerate when I say I find it frustrating to the read with the information removed. Without removing the text completely, just a `The `alt` property is important for the purpose of accessibility. Use alt="" if image is not a key part of page content.` would massively improve it imo.

For context, I based the work on errors on (weeks/months of research on) Elm and Rust, both projects widely acknowledged for their empowering and instructive errors and hints. Rust mostly achieve it through showing your own code, which we can't do a lot in Astro, whereas [Elm has quite literally full page errors crafting an entire story in the first-person around the error (it's a lot)](https://hub.packtpub.com/wp-content/uploads/2019/10/Elm-syntax.png). I believe a lot in the middle ground I reached, which I why I defend it a lot in this PR.

--- 

> ![image](https://github.com/withastro/astro/assets/3019731/f7beb5ef-1be4-4608-a8a4-3dc1b4246571)

To be honest, I kinda find all the information cramped like this to be very hard to read! Modern error reporting is now quite formatted, like this example from Miette: 
![image](https://github.com/withastro/astro/assets/3019731/76b6534e-34bd-4993-a9e1-27a6d5814b85) or Rust: 
![image](https://github.com/withastro/astro/assets/3019731/c970ed92-a4ab-4ce8-a89b-e05a9691b5d1)

Reply:
I wouldn't over-index on any one change made here, as I mentioned elsewhere I likely removed more than neccesary and am happy to workshop any of these individually (see specific improvements below).

If you'd like to keep using the "alt" example as representative of the rest of these changes, I'm happy to. Here's what I'd suggest now based on your first pass of feedback:

<img width="744" alt="Screen Shot 2023-10-20 at 11 29 07 AM" src="https://github.com/withastro/astro/assets/622227/2d85d184-d75c-42e1-974f-ba8ddd5a23ae">

vs. what we have today (reminder: in a window that I had to expand from VSCode's default size):

<img width="726" alt="Screen Shot 2023-10-20 at 11 35 30 AM" src="https://github.com/withastro/astro/assets/622227/c926c3ca-10bd-4312-8fb1-dc7ad0de5709">

The two pieces of information that have been removed:
1. **why "alt" is required:** The "why" doesn't materially change the error, how to solve, or the fact that alt is required. It is definitely interesting context, but unnecessary and verbose for people who already know about "alt", taking up about half of the error text/hint. In the new screenshot, if the user is curious to know why alt is required, they are still free to click the "learn more" link to get the answer.
2. **a link to the MDN docs on the "alt" attribute:** Putting aside that the `#attr-alt` URL hash is now broken and the impossibility of maintaining external links that we don't control, this information (and the direct link to MDN) are both already contained at the docs.astro.build URL. Showing two URLs makes it unclear where the user should go to learn more about the error.

Reply:
Also, please know that none of this is meant as a criticism of your work and I appreciate the discussion! Subjective changes like "make error pretty" are hard to talk through so I'm trying my best to distill to objective truths about what we do well today and what we could do better.

The fact that we even have hints and hosted error references in our docs at all is huge! Ironically, my intention here is to highlight both of these more than we do today. For ex, if the docs URL isn't the decisive resource to explain the error and the context behind it, we should improve the content on that page vs. pointing you somewhere else.

Reply:
Subjectively I'd like to have as much information in the terminal as possible. I don't want to have to flip over to a webpage to understand what I did wrong. But that also depends on how well I know what i'm doing, I take the point about conciseness being better if you already are aware of the problem. I'm not sure there is any one correct answer here.

What's unclear to me from this PR is if the motivation is about the error text or if its about fitting the message within a vscode window. If it were about a space concern there could be a short and long hint that gets displayed depending on the size of the terminal. But it's not clear to me that that's the true motivation, only that I've seen it brought up several times in this PR as a partial motivation.